### PR TITLE
Slurm infer

### PR DESCRIPTION
**Pull request by wbenoit26**
@deepchatterjeeligo Fairly large PR here. Does the following:

- Pins `torch` and `torchaudio` versions to v2.10.0 in `train`, `export`, `infer`, and `online`. This is a more modern version than we've been using, and the pinning makes sure that exported models are able to be opened by all projects.
- Replaces the old `torch.jit.save` and `torch.jit.load` with the newer `torch.export.export` and `torch.export.load`. The files produced by this use the `.pt2` format.
- Adds some formatting to the container-building script to allow per-project overwrites. This was necessary for what I wanted to do in the `infer` project.
- The significant piece: adds in-process inference as an alternative to using Triton. This is intended for clusters where there are a lot of GPUs available, but not necessarily many on the same node or where inter-node communication isn't possible. I tested it out on Delta, and ran through a year of background in ~1 hour using A40s, which is ~4x speed-up compared to DGX. Alongside this, I added a Delta profile. In principle, this can be used with OSG as well; all it would require is a new config in `pipeline/profiles/osg/` that mirrors the `ldg` profile, though I imagine we wouldn't want to doing anything besides inference on OSG.
- The in-process inference allows for three different backends: `export`, which just loads the `.pts` and runs data through it; `compile`, which compiles the model per-job and allows higher throughput at the cost of greater warmup time; and `aoti`, which uses ahead-of-time compilation to get higher throughput without the warmup cost, but requires compiling on the same GPU architecture as is used for inference.
- For both Triton and in-process inference, I added a `branches_per_job` parameter that allows you to tune how much `(file, shift)` combinations a single job processes. Useful because most branches can run quickly, so if you're compiling, you only have to pay once for the warm-up. For the run I did on Delta, I set 50 branches per job.

This is a large change, so merge at your leisure.

### CODE REVIEW

**Comment by wbenoit26** [comment]
Closes #496

**Review by deepchatterjeeligo — APPROVED** [review]
We discussed this at length on the call.
